### PR TITLE
Components: Remove ProgressIndicator Component

### PR DESCRIPTION
**Pull request by drw158**
It appears that this is not being used anywhere, and it’s not documented in the devdocs. Would be great to know what this component looks like though before removal. Looks like it hasn't been changed since the initial repo commit.

No visual changes.

A component of the same name resides in `components/wizard/progress-indicator` and is still being used by the wizard component.

Any help would be appreciated — I've never removed a component before.

### CODE REVIEW

File: client/components/progress-indicator/style.scss [hunk 1]
```diff
@@ -1,223 +0,0 @@
-// ==========================================================================
-// .progress-indicator
-//
-// Includs a spinner and the ability to have specific progress-indicator__percentages using the
-// same elements.
-//
-// Check out this demo to understand how this works:
-// http://codepen.io/MichaelArestad/pen/2740d550eaa89cef6c753316595298b7
```
Comment (vindl) [inline client/components/progress-indicator/style.scss]: > Would be great to know what this component looks like though before removal.

It seems like this demo partly answers that question.

**Comment by vindl** [comment]
> A component of the same name resides in components/wizard/progress-indicator and is still being used by the wizard component.

I've checked out the code for that one, and while it has the same name the functionality is different. Judging by the code the purpose of that one is to just display the step count, while this one should show a spinner.

File: client/components/progress-indicator/style.scss (hunk 1 above)
Comment (drw158) [reply client/components/progress-indicator/style.scss]: Ah! I missed that. I suppose the Spinner has replaced this component? https://wpcalypso.wordpress.com/devdocs/design/spinner

I'll cc @MichaelArestad

Comment (drw158) [reply client/components/progress-indicator/style.scss]: It seems I was mistaken, the Spinner is just a loading animation, and the progress indicator shows status messages.

I also missed that site indicator was using it, so we shouldn't remove the component.

**Comment by AnnaMag** [comment]
Wizard is used in Jetpack Onboarding, where ProgressIndicator marks the screen count.
Also, we are about to use it in the Jetpack remote install flow, again to mark progression.

@tyxla could you help judge its utility, please ? :)

**Comment by tyxla** [comment]
@AnnaMag the `ProgressIndicator` we're using is a different one - it's situated in the `Wizard` component and is specific to it.

As @vindl pointed out, this component is still used in the `SiteIndicator` component. To be able to see it, you need to have a Jetpack site with an older WP version (not the latest one). 

Then, selecting that site in Calypso will display an update indicator:



and clicking the update indicator will reveal the following:



where if you click the "Update to *.*.*" link, you will see the progress indicator shortly.

But note that there is a PR to remove this functionality from Calypso - #22617. Once removed, this component could be removed too.

My $0.02 - I'd prefer keeping this component and adding it to devdocs with an example and README. It could be useful in some future cases, unless there is a design reason not to have it, or we have a better alternative.

**Comment by drw158** [comment]
My main concern is the noticon that is being used by this component. After reading #22617, I think the component should be removed as well, because this is a pretty specific use case.

**Comment by MichaelArestad** [comment]
If it's not used anywhere or will be replaced by a spinner, I'm cool with removing it.

**Comment by MichaelArestad** [comment]
FWIW we never used it much. I think only in plugins and I think it might no longer be used there. It was a combination with a spinner, a smiley face (yes), a checkmark, and a cancel X if I remember correctly.

**Comment by tyxla** [comment]
#22617 has been merged, so this component is no longer used. It should be safe to remove now.

**Pull request merged**